Fix compiler warnings

mkbel opened the pull request.

mkbel committed:
Fix compiler warning: sketch/Marlin.h:151:25: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]

mkbel committed:
Fix 4 compiler warnings of type: sketch/planner.cpp:872:59: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]

mkbel committed:
Fix compiler warning sketch/Marlin_main.cpp:6584:19: warning: comparison is always true due to limited range of data type [-Wtype-limits]

mkbel committed:
Remove unused function, fix compiler warning: sketch/Marlin_main.cpp:8994:6: warning: unused parameter 'extruder' [-Wunused-parameter]

mkbel committed:
Fix compiler warnings: sketch/cardreader.cpp: In member function 'void CardReader::openFile(const char*, bool, bool)':

sketch/cardreader.cpp:350:25: warning: ordered comparison of pointer with integer zero [-Wextra]
sketch/cardreader.cpp:355:22: warning: ordered comparison of pointer with integer zero [-Wextra]

mkbel committed:
Fix stack corruption for folder name longer than 12 characters. Save 260B of flash memory. Move duplicate code to separate method. Fix compiler warnings: sketch/cardreader.cpp:448:25: warning: ordered comparison of pointer with integer zero [-Wextra]

sketch/cardreader.cpp:453:22: warning: ordered comparison of pointer with integer zero [-Wextra]

mkbel committed:
Do not compare signed value with unsigned, add parentheses.

3d-gussner commented:
The warnings in 1.6.9 are gone. And with this one even got rid of some in 1.8.5 ... there are still some :-;
Thanks @mkbel great job cleaning up the code

mkbel committed:
Fix compiler warning: sketch/mesh_bed_calibration.cpp:135:20: warning: unused parameter 'i' [-Wunused-parameter]

mkbel committed:
Fix compiler warning: sketch/mesh_bed_calibration.cpp:205:34: warning: unused parameter 'verbosity_level' [-Wunused-parameter]

mkbel committed:
Fix compiler warning: sketch/mesh_bed_calibration.cpp:966:13: warning: unused parameter 'verbosity_level' [-Wunused-parameter]

mkbel committed:
Fix compiler warning: sketch/mesh_bed_calibration.cpp:1057:13: warning: unused parameter 'verbosity_level' [-Wunused-parameter]

mkbel committed:
Fix compiler warnings sketch/mesh_bed_calibration.cpp:150:21: warning: unused parameter 'npts' [-Wunused-parameter]

sketch/mesh_bed_calibration.cpp:172:21: warning: unused parameter 'npts' [-Wunused-parameter]

mkbel committed:
Fix compiler warning: sketch/mesh_bed_leveling.cpp:24:6: warning: unused parameter 'use_default' [-Wunused-parameter].

mkbel committed:
Fix compiler warning sketch/mmu.cpp:32:6: warning: unused parameter 'extr' [-Wunused-parameter]

mkbel committed:
Save 8B flash, fix compiler warning sketch/sound.cpp:62:6: warning: unused parameter 'eSoundClass' [-Wunused-parameter]

mkbel committed:
Fix compiler warning sketch/ultralcd.cpp:53:31: warning: missing initializer for member 'MenuData::BabyStep::babystepMem' [-Wmissing-field-initializers]

mkbel committed:
Save 278B of flash. Remove unused function parameters. Make functions static. Fix compiler warnings:

sketch/ultralcd.cpp:292:6: warning: unused parameter 'pstr' [-Wunused-parameter]
sketch/ultralcd.cpp:292:6: warning: unused parameter 'filename' [-Wunused-parameter]
sketch/ultralcd.cpp:344:6: warning: unused parameter 'pstr' [-Wunused-parameter]
sketch/ultralcd.cpp:364:6: warning: unused parameter 'pstr' [-Wunused-parameter]
sketch/ultralcd.cpp:385:6: warning: unused parameter 'pstr' [-Wunused-parameter]
sketch/ultralcd.cpp:6920:6: warning: unused parameter 'longFilename' [-Wunused-parameter]
sketch/ultralcd.cpp:6965:6: warning: unused parameter 'longFilename' [-Wunused-parameter]

mkbel committed:
Fix showing calibration results.

Messages:
_i("XYZ calibration failed. Bed calibration point was not found.")
_T(MSG_BED_SKEW_OFFSET_DETECTION_FITTING_FAILED);
_i("XYZ calibration failed. Front calibration points not reachable.")
_i("XYZ calibration failed. Right front calibration point not reachable.")
_i("XYZ calibration failed. Left front calibration point not reachable.");
were never shown.
Fix compiler warnings:
sketch/ultralcd.cpp:3476:19: warning: comparison is always false due to limited range of data type [-Wtype-limits]
sketch/ultralcd.cpp:3478:26: warning: comparison is always false due to limited range of data type [-Wtype-limits]

mkbel committed:
Fix compiler warning: sketch/stepper.cpp:1499:6: warning: unused parameter 'address' [-Wunused-parameter].

mkbel committed:
Fix compiler warning sketch/stepper.cpp:1553:6: warning: unused parameter 'driver' [-Wunused-parameter]

mkbel committed:
Save 32B of FLASH, fix compiler warning: sketch/tmc2130.cpp:238:6: warning: unused parameter 'last_step_mask' [-Wunused-parameter].

mkbel committed:
Fix compiler warning: sketch/xyzcal.cpp:90:6: warning: unused parameter 'de' [-Wunused-parameter].

mkbel committed:
Fix compiler warnings: sketch/xyzcal.cpp:111:10: warning: unused parameter 'nd' [-Wunused-parameter]

sketch/xyzcal.cpp:111:10: warning: unused parameter 'dd' [-Wunused-parameter]

mkbel committed:
Fix compiler warnings: sketch/uart2.c:19:33: warning: unused parameter 'stream' [-Wunused-parameter]

sketch/uart2.c:28:25: warning: unused parameter 'stream' [-Wunused-parameter]
sketch/Marlin_main.cpp:925:5: warning: unused parameter 'stream' [-Wunused-parameter]
sketch/lcd.cpp:224:5: warning: unused parameter 'stream' [-Wunused-parameter]

mkbel committed:
Remove factory_reset() unused quiet parameter and make it static. It was tested, that both calls of factory reset (from menu and via PRUSA FR command) works somehow. Fix compiler warning: sketch/Marlin_main.cpp:808:6: warning: unused parameter 'quiet' [-Wunused-parameter]

mkbel committed:
Supress compiler warning sketch/Marlin_main.cpp:3079:6: warning: unused parameter 'e_shift_late' [-Wunused-parameter]. Make function static. Fix indentation.
